### ISSUE TITLE
Modification of values in input image based on annotation's mask

### ISSUE THREAD

**Issue by mpelchat04** [NRCan/geo-deep-learning#85]
In [image_to_samples.py](https://github.com/NRCan/geo-deep-learning/blob/e443f67de6b0113ce5b2773ee4712cd018495925/images_to_samples.py#L204), it was possible to modify values in input image based on annotation's mask. It was used when the extent of the input image was bigger than annotation's extent.   
Since [Pytorch 0.4.1](https://pytorch.org/docs/0.4.1/nn.html#torch.nn.functional.cross_entropy), loss functions support `ignore_index`. When specified, the value in `ignore_value` isn't computed during loss calculation. This should be used instead of the current behavior. Moreover, `ignore_index` should be include in the config file, as a training hyperparameter.
